apps/frontend/src/lib/oauth/imports.ts

```diff
@@ -28,6 +34,63 @@ async function stravaFetch(token: AccessToken): Promise<Response> {
   );
 }
 
+async function ghFetch(token: AccessToken): Promise<Response | null> {
+  const user = await fetch(
+    "https://api.github.com/user",
+    {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": `Bearer ${token.access_token}`,
+      },
+    }
+  );
+
+  const rawUser = await user.json();
+
+  const resp: GithubUserResponse = GithubUserResponseSchema.parse(rawUser);
+
+  const username = resp.login;
+  // NOTE: with the exception of year, values are indices
+  const from = new Date(2024, 9, 21); // Oct 20, 2024
+  const to = new Date(2024, 9, 28); // Oct 27, 2024
+
+  // TODO: find graphgl library for handling query / variables
```

Comment:
I should make this safer / easier to deal with tomorrow by finding a graphql ts library.